src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php

```diff
@@ -362,6 +362,83 @@ private function getDenormalizerForObjectWithBasicProperties()
         return $denormalizer;
     }
 
+    public function testDenormalizeBasicTypePropertiesWithStringBoolValues()
```

Comment:
Wouldn't it be better robuste a dataProvider here? The getDenormalizerForObject...() is tight coupled to the number of values provided by the on consecutive calls, while it's not supposed to be extended in the future I guess, it feels a bit weird.

While using a dataProvider it would be less complex and you don't the onConscutiveCalls anymore

Reply:
If this was my own library I would have written tests differently, but I tried to keep with the same style the all other tests are done in the file.

https://github.com/symfony/symfony/blob/8e6b1f836c9173532189a3dc2b1677e3c29dd5c4/src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php#L292-L440